### PR TITLE
Rename Variable, Record and Array (#363)

### PR DESCRIPTION
**Pull request by sveinse**
To create better distinction between the different Variable, Record and Array types used for different purposes. Helps development, as IDE/linters often only display base name of class. Discussed in issue #363 

Unittests pass and generated sphinx documents have been skimmed over.

### CODE REVIEW

**Review by christiansandberg — APPROVED** [review]
Seems like a good improvement. As long as we keep backwards compatibility I see no problem.

**Comment by sveinse** [comment]
Updated with lastest commits in master.

File: canopen/pdo/base.py [hunk 1]
```diff
@@ -528,11 +528,11 @@ def wait_for_reception(self, timeout: float = 10) -> float:
         return self.timestamp if self.is_received else None
 
 
-class Variable(variable.Variable):
+class PdoVariable(variable.Variable):
```
Comment (acolomb) [inline canopen/pdo/base.py:531]: This is pure bikeshedding, but why do you use CamelCase with the lowercase `Pdo` abbreviation here, while for the OD you use the all-caps `OD` spelling as in `ODVariable`?  Shoult consistenly use uppercase for abbreviations here.

File: canopen/sdo/base.py [hunk 2]
```diff
@@ -80,14 +80,14 @@ def download(
         raise NotImplementedError()
 
 
-class Record(Mapping):
+class SdoRecord(Mapping):
```
Comment (acolomb) [inline canopen/sdo/base.py:83]: Same here, `SDORecord`?  Please just be consistent.

**Review by acolomb — CHANGES_REQUESTED** [review]
Looks good in principle, just a nit-pick about inconsistent uppercase variants.

File: canopen/pdo/base.py (hunk 1 above)
Comment (sveinse) [reply canopen/pdo/base.py:531]: @acolomb you're right, this is inconsistent. I didn't make a very deliberate choice, rather trying to adopt the gist from the rest of canopen. After looking more closer at it, I see canopen is equally inconsistent, e.g. `class PDO` vs `class PdoBase`. Looking at all class names, the majority is CamelCase, with the notable exception of `PDO`, `TPDO`, `RPDO` and not used with an additional suffix.

To be consistent with naming in canopen, `ODVariable` should then be `OdVariable` perhaps?

File: canopen/sdo/base.py (hunk 2 above)
Comment (sveinse) [reply canopen/sdo/base.py:83]: With reference to comment above, `SdoRecord` is correct if `SdoBase` is correct I would argue.

File: canopen/pdo/base.py (hunk 1 above)
Comment (acolomb) [reply canopen/pdo/base.py:531]: Yes, I guess that's an acceptable solution.  Just don't introduce even more inconsistency ;-)

Comment (christiansandberg) [reply canopen/pdo/base.py:531]: Naming is probably inconsistent everywhere else anyway, so it's not super important. Not sure what's best for readability, but I'm nowadays leaning more towards keeping abbreviations upper-case even if it makes it harder to separate the words.

**Pull request merged**